src/js/services/dialer.js

```diff
@@ -34,6 +35,44 @@ export default App.extend({
   call(number, action) {
     this._call(number, action);
   },
+  five9CallNumber(values) {
+    if (!values) {
+      patients.reset();
+      return;
+    }
+
+    const { actionId, number } = values;
+    const currentUrl = window.location.href;
+    const action = Radio.request('entities', 'actions:model', actionId);
+
+    if (action) {
+      const patient = action.getPatient();
+
+      if (patient) {
+        if (!patients.get(patient.id) && !currentUrl.includes(patient.id)) {
```

Comment:
Don't add the patient button if the patient is already being shown or the user is already on that patient's dashboard (via url containing patient id).

Reply:
I'm starting to wonder if we should just always add the patients to the collection and then just use a viewFilter on the collection that does this url check against the id.  that's probably cleaner..  don't have to worry about when to add/remove.  FWIW we can even just always add.. the collection will prevent duplicates